wordpress-mobile/WordPress-Android
# [Gutenberg] File Block -  FileBlockProcessor for upload media url transformation

## Findings
This PR is the Android side of the [iOS implementation for updating the HTML content](https://github.com/wordpress-mobile/WordPress-iOS/pull/15339) of the File Block without Gutenberg being open. 

## Solution
The `FileBlockProcessor` replaces all the `<a href>` that contain local file URLs with remote media URLs once the file upload is complete. The id of the File Block is also updated with a remote variant as well. 

@malinajirka while working on this I saw this use case [CleanUpMediaToPostAssociationUseCase](https://github.com/wordpress-mobile/WordPress-Android/blob/7ed34a51eab27dd77aa1b87b948bc296b478deb8/WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/CleanUpMediaToPostAssociationUseCase.kt) 

Since a file is basically a media item should we modify the regex for finding media related blocks below to include the file block so the use case mentioned above is run for it as well? 

https://github.com/wordpress-mobile/WordPress-Android/blob/e693397bc0cb879ec4fc8ea4efeb6114bb6dd44d/WordPress/src/main/java/org/wordpress/android/ui/posts/PostUtils.java#L444

## Testing

1. Add a file block to a new post.
2. Upload a file to the block (I  lessened the Cellular Network type and Signal strength to replicate a slow network so I  had enough time to close the editor).
3. Close the editor by pressing the back button.
4. Let the upload to finish in the background.
5. When the file is uploaded and the post saved, open the post again (on web or mobile).

Verify that the behavior is functioning correctly by :
  - Checking that the block renders properly.
  - Checking that all references to href are updated to the remote URL.
  - Checking that the file ID has been updated to the remote one.

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
